tests/regression_functional_test.go

```diff
@@ -0,0 +1,122 @@
+//// name : path_to_yml
+var osDict = map[string]string{
+	"testvm-fedora" : "manifests/vm-template-fedora.yml",
+}
+
+const (
+	temporaryJson = "/tmp/tmp-vm.yml"
+	ipQuery = "-o=jsonpath='{.status.interfaces[0].ipAddress}'"
+	phaseQuery = "-o=jsonpath='{.status.phase}'"
+)
+
+
+var _ = Describe("Regression and Functional tests of VMs and VMIs", func() {
+
+	flag.Parse()
+
+	Describe("Creating VMs with different OS from templates via oc and kubectl", func() {
+
+		// Run tests for each OS:
+		for k := range osDict {
+
+			osName := k
+			templatePath := osDict[k]
+
+			Context("Create/Delete " + osName + " VM from template via oc command", func() {
+
+				It("Create " + osName + " VM, start and check status", func() {
+					By("Create VM")
+					tests.ProcessTemplateWithParameters(templatePath, temporaryJson, "NAME="+osName, "CPU_CORES=2")
+					tests.CreateResourceWithFilePathTestNamespace(temporaryJson)
+
+					By("Start VM")
+					_, _, err := tests.StartStopVirtualMachineVirtctl(osName, "start")
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Waiting until VMI is running")
+					tests.WaitUntilResourceReadyByNameTestNamespace("vmi", osName, phaseQuery, "Running")
+
+					ipAddress := tests.GetVirtualMachineSpecificParameters("vmi", osName, ipQuery)
+					// removing quotes
+					ipAddress = ipAddress[1 : len(ipAddress)-1]
+
+					By("Waiting for the console and check IP address (now only for Fedora)")
+					if osName == "testvm-fedora" {
+						expecter, err := tests.LoggedInFedoraExpecter(osName, tests.NamespaceTestDefault, 300)
+						Expect(err).ToNot(HaveOccurred())
+						defer expecter.Close()
+
+						_, err = expecter.ExpectBatch([]expect.Batcher{
+							&expect.BSnd{S: "ifconfig\n"},
+							&expect.BExp{R: ipAddress},
+						}, 120*time.Second)
+						Expect(err).ToNot(HaveOccurred())
+					}
+				})
+
+				It("Stop and delete " + osName + " VM'", func() {
+					By("Stop VM")
+					_, _, err := tests.StartStopVirtualMachineVirtctl(osName, "stop")
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Delete VM")
+					tests.DeleteResourceWithFilePathTestNamespace(temporaryJson)
+
+					By("Verify that VM was removed")
+					tests.WaitUntilResourceDeleted("vm", osName)
+				})
+			})
+
+			Context("Create/Delete " + osName + " VM from template via kubectl command", func() {
+
+				It("Create " + osName + " VM, start and check status", func() {
+					By("Create VM")
+					tests.ProcessTemplateWithParameters(templatePath, temporaryJson, "NAME="+osName, "CPU_CORES=2")
+					tests.CreateResourceWithFilePathTestNamespaceViaKubectl(temporaryJson)
+
+					By("Start VM")
+					_, _, err := tests.StartStopVirtualMachineVirtctl(osName, "start")
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Waiting until VMI is running")
+					tests.WaitUntilResourceReadyByNameTestNamespace("vmi", osName, "-o=jsonpath='{.status.phase}'", "Running")
+				})
+
+				It("Stop and delete " + osName + " VM'", func() {
```

Comment:
same

Reply:
Tests combined runs in loop (for for oc and kubectl)